hermes/file_gen.go

```diff
@@ -42,10 +42,13 @@ func (file HermesFile) generateFileChecksum() string {
 }
 
 // method of HermesFile generates the file takes id as a parameter
-func GenerateHermesFile(id int) HermesFile {
+func NewHermesFile(id int) (*HermesFile, error) {
```

Comment:
It might also be worth keeping checking somewhere whether the requested id has already been taken

Reply:
That will be done within the createFile probe! We will call NewHermesFile(id) for a 'missing' id from the list